Toolkit/ArcGISToolkit/TemplatePickerViewController.swift

```diff
@@ -0,0 +1,296 @@
+                continue
+            }
+            
+            fl.load{ [weak self] error in
+                
+                if error != nil{
+                    return
+                }
+                
+                guard let strongSelf = self else{
+                    return
+                }
+                
+                guard let popupDef = fl.popupDefinition, popupDef.allowEdit || table.canAddFeature else{
+                    return
+                }
+                
+                let tableTemplates = table.featureTemplates.map({
+                    FeatureTemplateInfo(featureLayer:fl, featureTable:table, featureTemplate:$0, swatch:nil)
+                })
+                
+                let typeTemplates = table.featureTypes
+                    .flatMap({ $0.templates })
+                    .map({ FeatureTemplateInfo(featureLayer:fl, featureTable:table, featureTemplate:$0, swatch:nil) })
+                
+                let infos = tableTemplates + typeTemplates
+                
+                // add to list of unfiltered infos
+                strongSelf.unfilteredInfos.append(contentsOf: infos)
+                
+                // re-assign to the current infos so we can update the tableview
+                // only should do this if not currently filtering
+                if !strongSelf.isFiltering{
+                    strongSelf.currentInfos = strongSelf.unfilteredInfos
+                }
+                
+                // generate swatches for the layer infos
+                for (index, info) in infos.enumerated(){
+                    if let feature = info.featureTable.createFeature(with: info.featureTemplate){
+                        let sym = info.featureLayer.renderer?.symbol(for: feature)
+                        sym?.createSwatch{ image, error in
+                            
+                            if error != nil{
+                                return
+                            }
+                            
+                            // update info with swatch
+                            infos[index].swatch = image
+                            
+                            // reload index where that info currently is
+                            if let index = strongSelf.indexPathForInfo(info){
+                                strongSelf.tableView.reloadRows(at: [index], with: .automatic)
+                            }
+                        }
+                    }
+                }
+            }
+            
+        }
+        
+    }
+    
+    // MARK: TableView delegate/datasource methods
+    
+    public func numberOfSectionsInTableView(_ tableView: UITableView) -> Int{
+        return tables.count
+    }
+    
+    public func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return tables.count == 0 ? nil : tables[section].tableName
```

Comment:
The table view only calls this after getting the number of sections from you. If you tell it there are zero sections, it won't ask you for the title for the header in any section. Therefore it is fine to omit the check for `tables.count` and just return `tables[section].tableName`. Does that make sense?

Reply:
yes - thanks